dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/patch/ItemPatch.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.repository.patch;
+
+import java.sql.SQLException;
+import java.util.UUID;
+
+import org.apache.log4j.Logger;
+import org.dspace.app.rest.exception.PatchBadRequestException;
+import org.dspace.app.rest.exception.PatchUnprocessableEntityException;
+import org.dspace.app.rest.model.patch.Operation;
+import org.dspace.app.rest.model.patch.Patch;
+import org.dspace.authorize.AuthorizeException;
+import org.dspace.content.Item;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the implementation for item resource patches.
+ *
+ * @author Michael Spalti
+ */
+@Component
+public class ItemPatch extends AbstractResourcePatch {
+
+    private static final String OPERATION_PATH_WITHDRAW = "/withdrawn";
+
+    private static final String OPERATION_PATH_DISCOVERABLE = "/discoverable";
+
+    private static final Logger log = Logger.getLogger(ItemPatch.class);
+
+    @Autowired
+    ItemService is;
+
+    @Override
+    public void replace(Context context, String apiCategory, String model, UUID id, Patch patch, Operation operation)
+        throws PatchUnprocessableEntityException, PatchBadRequestException, SQLException, AuthorizeException {
+
+        Item item;
+        try {
+            item = is.find(context, id);
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
+            throw e;
+        }
+
+        switch (operation.getPath()) {
+            case OPERATION_PATH_WITHDRAW:
+                withdraw(context, item, (Boolean) operation.getValue());
+                break;
+            case OPERATION_PATH_DISCOVERABLE:
+                discoverable(item, (Boolean) operation.getValue());
+                break;
+            default:
+                throw new PatchUnprocessableEntityException(
+                    "Unrecognized patch operation path: " + operation.getPath()
+                );
+        }
+    }
+
+    /**
+     * Withdraw or reinstate item based on boolean value provided by the patch request.
+     *
+     * The issue (https://jira.duraspace.org/browse/DS-3909) asks the operation to return the 422 error code
```

Comment:
This is not implemented because I'm not sure what to test for. What does it mean for something not to be archived?  Should I check the "isArchived" flag in the item or is there something else that I should look for?